javascript/javascript1/week2/student-manager.js

```diff
@@ -0,0 +1,58 @@
+
+const class07Students = [];
+function addStudentToClass(studentName) {
+    if (studentName) {
+        if (class07Students.length < 6) {
+            if (class07Students.includes(studentName)) {
+                console.log(`Student ${studentName} is already in the class`)
+                }
+            else {
+                return class07Students.push(studentName)
+                } 
+        }else if (studentName === 'Queen'){
+            return class07Students.push('Queen')
+        }else {
+            console.log('Cannot add more students to class 07') 
+            }
+        }
+   
+}
```

Comment:
Leverage that you can `return` at any point in a function (which will avoid the rest of the function from being executed). Doing this can greatly reduce nesting of `if` conditions and in turn make the function easier to read. You need to be smart about the order in which checks are done though. I'll begin with the Queen because that is a rule that beats all others:
```js
function addStudentToClass(studentName) {
  if (!studentName) {
    return
  }

  ...
```

You might want to add a `console.log` to show that student name needs to be provided, but your version of `addStudenToClass` didn't so I left it out in the above example to match your implementation.